### DISCUSSION TITLE
Is it possible to generate paragraphes of real English (& other Natural Languages)

### DISCUSSION THREAD

**Discussion by ghost** [faker-js/faker#702] [Q&A]
Hi,
Thanks a lot for providing and maintaining this library.

Is if possible to generate English (or other languages, I know about Localization..) text, at least English worlds with no particular semantic meaning. So another option in addition to "lorem.slug"

If this is not possible yet, I think it is really a good use case, basically adding dictionaries and randomly pick words (but I know how you do good stuff :))

**Comment by xDivisionByZerox** [comment]
From my understanding, this is currently not possible. But one of the maintainers can probably give some more insights about this.

For your own use case, you can build sentences yourself like:
```
import faker from '@faker-js';

const text = `Hello I'm ${faker.name.firstName()} ${faker.name.lastName()} and I'm ${faker.datatype.number(50)} years old.
Btw, I was born in ${faker.address.city()}.
If you want to get in contact you can write me under "${faker.internet.email()}".`;

console.log(text);
// Hello I'm Nellie Hettinger and I'm 11 years old.
// Btw, I was born in Alexandria.
// If you want to get in contact you can write me under "Guy23@yahoo.com".
```

**Reply by ghost** [reply]
Thanks @xDivisionByZerox It can be a temporary solution as to combine some random: lastnames, firstnames, ...
I think it is really useful to provide a new Module, It would just pick randomly from dictionaries.
thanks let's wait for other maintainers.

**Reply by xDivisionByZerox** [reply]
The problem with your request is that language itself is not as simple as putting a random number of blocks together and having a valid result. A solution to this problem could be a completely new library itself.

A module with very basic sentences (like I have given) could become a module in the future I guess. 
But that could get repetitive really fast IMO.

**Reply by ghost** [reply]
humm, I remember reading about the concept of a "word" in natural language, but forgot about the technical term of a "word" as in English. But it still think all languages have such concept.

**Reply by xDivisionByZerox** [reply]
Okay so maybe I misunderstood what you wanted to do. For simplicity ignore localization for now.
Do you want just any English **WORD** at all like 'tree', 'school', 'apple' OR do you want to create **SENTENCES** like 'I am eating an apple.'?

**Reply by ST-DDT** [reply]
**Why** do you need "real" English sentences? What is your usecase?

We sure could generate some theoretically valid English sentences, but they would be kind of random and out of context (and they would probably be more agonizing to read, then simple lorem texts).

````txt
Foobar is fishing in the forest.
Jennifer is skiing on a skateboard.
In 3 days John is hacking on a accomplishment.`
````

If you really need "meaningful" text you might be better off using some actual text generators like GPT-3 or similar tools.

**Reply by ghost** [reply]
@xDivisionByZerox  Sentences which are a combination of random words with  a short memory of words. 

Basically my problem is I'm having random posts with a title (sentence) and a content (paragraph) and also tags (words) and I wish to have some ressemblance between tags and titles. 

The repetition of a word is so minimal in "lorem.slug" which is not like in natural languages. 

Ok, now I have an idea, let's not think of my suggestion, let's keep "lorem.slug" and be able to add entropy (basically repetition of the same word in a sentence or in a paragraph),

**Reply by ghost** [reply]
@ST-DDT  true, no need for real English I stated English words with no semantic meaning.  (I agree on the ambiguity that this would result to).
But check my last reply for why I'm needing this.

**Reply by ghost** [reply]
I just figured out there is seed variable. I can generate paragraphs from words passing same seeds from time to time randomly... I don't know if this is fine, Maybe this is a temporary solution for me, I will try this later..

**Reply by ST-DDT** [reply]
So basically you want something like this?

> **Jennifer likes fishing**
>
> Aut **fishing** **likes** cupiditate **Jennifer** magnam quia aliquid est. At impedit **Jennifer** numquam **fishing**. Voluptates **likes** **fishing** illo magni **Jennifer** exercitationem quam cupiditate enim. Et atque totam **Jennifer** quasi unde rerum officiis. 
> Et et nihil **likes** accusantium **Jennifer** veniam suscipit molestias perferendis. Unde quo corrupti dolorem perferendis. Iste molestiae **Jennifer** minima **fishing** cumque adipisci. Accusamus **likes** autem aliquid voluptatibus cupiditate qui illum qui. 
> Ea nemo eum ab aut **likes** cumque sed velit **fishing** ducimus quam. Est et **likes** consequuntur **Jennifer** occaecati dolorum id **likes** **likes** et nulla **Jennifer** ipsam. Atque sit quasi deserunt **Jennifer** dolores rem. Rerum veniam minima. Aut **fishing** consectetur **Jennifer** placeat **likes** odio id non consequatur **Jennifer** officia nisi.

(Maybe with random English words instead of Lorem ones?)

**Reply by ghost** [reply]
At least some repetitions. In words all over but yes that is a nice example. I don't know how Lorem texts are generated, but for my case I nearly see no repetitions. 
I'm using `json-schema-faker` which uses Faker, and this is a demo https://classified-ads-bacloud14.herokuapp.com/ check for instance listings: titles, description, and tags. For my development journey, (would not be used in prod) it would be nice to have some repetitions ( concept of entropy, not that hard)

Thanks for following up 🙂

**Reply by ghost** [reply]
For now, I'm just picking random words from a prebuild real dictionary for different languages. All if fine for me. 

Thank you for the follow up
